src/pages/login-page/LoginPage.js

```diff
@@ -0,0 +1,164 @@
+import { React, useState } from "react";
+import "./loginpage.css";
+import { Link, useNavigate } from "react-router-dom";
+import axios from "axios";
+import { useAuth } from "../../contexts";
+import { ToastContainer, toast } from "react-toastify";
+import "react-toastify/dist/ReactToastify.css";
+import bcyrpt from "bcryptjs";
+
+export function LoginPage() {
+  const [type, setType] = useState("password");
+  const [userData, setUserData] = useState({
+    email: "",
+    password: "",
+  });
+  let navigate = useNavigate();
+  const { setUser, setIsLoggedIn } = useAuth();
+  const [error, setError] = useState("");
+  const [passwordError, setPasswordError] = useState("");
+
+  const doValidate = () => {
+    if (
+      !/^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/.test(
+        userData.email
+      )
+    ) {
+      setError("Please enter a valid email address");
+      return false;
+    } else {
+      setError("");
+    }
+    if (
+      userData.password === "" ||
```

Comment:
suggestion you could also do !userData.password if will do the same.

Reply:
replaced.